hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/security/OzoneDelegationTokenSecretManager.java

```diff
@@ -528,6 +532,10 @@ private void addPersistedDelegationToken(OzoneTokenIdentifier identifier,
     byte[] password;
     if (StringUtils.isNotEmpty(identifier.getSecretKeyId())) {
       ManagedSecretKey signKey = secretKeyClient.getSecretKey(UUID.fromString(identifier.getSecretKeyId()));
+      if (signKey == null) {
```

Comment:
@jojochuang , thanks for working on this improvements.  For these token whose sign key is not available, we'd better put them in a map/list, and delete them at then end of  loadTokenSecretState. 

BTW, do you have idea how long a secret key will be deleted from SCM after it's creation?

Reply:
Secret key default life time is 1 day.

Reply:
Update: secret life time is 7 days. hdds.secret.key.expiry.duration
there's a property hdds.secret.key.rotate.duration which let SCM create a new secret key every 1 day. (SCM will keep a few secret keys in memory)
The SCM background checks for expired secret key every 10 minutes.

Reply:
> I'd suggest to let the OM background thread (OzoneDelegationTokenSecretManager) to take care of expired dt later.

It makes sense.  Besides, for those dt whose fetcing secret key failed and also it is expired, those dt will be not put int memory map, so will not cleaned up by background thread, we delete them directly during load, thoughts?